### PR TITLE
add: ProgressBar component

### PR DESCRIPTION
**Pull request by EdPrys**
fixed: #6

Add: 
- component ProgressBar
- ProgressBar styles

### CODE REVIEW

File: components/ProgressBar/ProgressBar.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    const handleMouseUp = () => {
+      setIsDragging(false);
+    };
+
+    window.addEventListener("mouseup", handleMouseUp);
+
+    return () => {
+      window.removeEventListener("mouseup", handleMouseUp);
+    };
+  }, []);
```
Comment (Security2431) [inline components/ProgressBar/ProgressBar.tsx:26]: It is correct usage for event handlers, however I would recommend using the native react.js methods

**Review by Security2431 — COMMENTED** [review]
BTW, I'd recommend using dash naming in the branch

https://tilburgsciencehub.com/building-blocks/collaborate-and-share-your-work/use-github/naming-git-branches/

File: components/ProgressBar/ProgressBar.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+  const handleMouseMove = (event: React.MouseEvent<HTMLDivElement>) => {
+    if (isDragging && progressBarRef.current) {
+      const progressBarRect = progressBarRef.current.getBoundingClientRect();
+      const progressBarWidth = progressBarRect.width;
+      const offsetX = event.clientX - progressBarRect.left;
+      const progressPercentage = offsetX / progressBarWidth;
+
+      const clampedPercentage = Math.max(0, Math.min(1, progressPercentage));
+      const newValue = Math.round(clampedPercentage * 100);
```
[the remark this answers is not in the document]
Comment (EdPrys) [reply components/ProgressBar/ProgressBar.tsx]: fixed

File: components/ProgressBar/ProgressBar.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  const handleIconMouseDown = (event: React.MouseEvent<HTMLDivElement>) => {
+    event.stopPropagation();
+    setIsDragging(true);
+    setDragStartX(event.clientX);
+    setDragStartValue(value);
+  };
+
+  const calculatePinPosition = (event: React.MouseEvent<HTMLDivElement>) => {
+    const progressBarRect = progressBarRef.current?.getBoundingClientRect();
+    const progressBarWidth = progressBarRect?.width || 0;
```
Comment (Security2431) [inline components/ProgressBar/ProgressBar.tsx:25]: If we divide by zero, we get `Infinity`. Not sure we wanted infinity progress bar width 😅

File: components/ProgressBar/ProgressBar.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    const offsetX = event.clientX - (progressBarRect?.left || 0);
+    const progressPercentage = offsetX / progressBarWidth;
+
+    const clampedPercentage = Math.max(0, Math.min(1, progressPercentage));
+    const newValue = Math.round(clampedPercentage * 100);
```
Comment (Security2431) [inline components/ProgressBar/ProgressBar.tsx:30]: NIT: It's just about readability :) 
```suggestion
    const { left, width } = progressBarRef.current?.getBoundingClientRect() ?? {};

    const offsetX = event.clientX - left;
    const progressPercentage = offsetX / width;

    const clampedPercentage = Math.max(0, Math.min(1, progressPercentage));
    const newValue = Math.round(clampedPercentage * 100);
```

File: components/ProgressBar/ProgressBar.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+    return newValue;
+  };
+
+  const handleMouseMove = (event: React.MouseEvent<HTMLDivElement>) => {
+    if (isDragging && progressBarRef.current) {
+      const newValue = calculatePinPosition(event);
+      onChange(newValue);
+    }
```
Comment (Security2431) [inline components/ProgressBar/ProgressBar.tsx:39]:
```suggestion
    if (!isDragging || !progressBarRef.current) {
      return;
    }
    
    const newValue = calculatePinPosition(event);
    onChange(newValue);
    
```

**Comment by Security2431** [comment]
This PR is deprecated, we decided to move with `shaka-player` library

**Pull request closed without merging**